File: java/program14.java
import java.util.*;

class Array
{
    public int Arr[];
    public int icnt;

    public Array(int iSize)
    {
        Arr = new int[iSize];
        icnt = 0;
    }

    public void Accept()
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter elements of the array");

        for(icnt = 0; icnt < Arr.length; icnt++)
        {
            Arr[icnt] = sobj.nextInt();
        }
    }

    public void Display()
    {
        System.out.println("Elements of the array are :");

        for(icnt = 0; icnt < Arr.length; icnt++)
        {
            System.out.print(Arr[icnt]+"\t");
        }   
        System.out.println();
    }
}

class MyArray extends Array
{
    public MyArray(int iSize)
    {
        super(iSize);
    }

    boolean CheckNumber(int iSearch)
    {
        for(icnt = 0; icnt < Arr.length; icnt++)
        {
            if(Arr[icnt] == iSearch)
            {
                break;
            }
        }
        if(icnt == Arr.length)
        {
            return false;
        }
        else
        {
            return true;
        }
    }
}

class program14
{
    public static void main(String arg[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number of elements :");
        int iValue = sobj.nextInt();

        MyArray obj = new MyArray(iValue);
        obj.Accept();
        obj.Display(); 

        System.out.println("Enter number you want to search");
        int iNo = sobj.nextInt();

        boolean bRet = obj.CheckNumber(iNo);
        if(bRet == true)
        {
            System.out.println("Number is present");
        }
        else
        {
            System.out.println("Number is NOT present");
        }
    }
}

File: cpp/program14.cpp
// program88.c

#include<iostream>
using namespace std;

class Array
{
    public:
       int iSize;
       int *Arr;
    
    Array(int iNo)
    {
        iSize = iNo;
        Arr = new int[iSize];
    }

    void Accept()
    {
        cout<<"Enter elements of Array"<<"\n";

        int iCnt = 0;
        for(iCnt = 0; iCnt < iSize; iCnt++)
        {
            cin>>Arr[iCnt];
        }
    }

    ~Array()
    {
        delete []Arr;
    }

    bool SearchNum()
    {
        int iCnt = 0;
        int iSearch = 0;

        cout<<"Enter the element you want to search"<<'\n';
        cin>>iSearch;

        for(iCnt = 0; iCnt < iSize; iCnt++)
        {
            if(Arr[iCnt] == iSearch)
            {
                break;
            }
        }
        if(iCnt == iSize)
        {
            return false;
        }
        else
        {
            return true;
        }
    }

};

int main()
{
    int iValue = 0;
    bool bRet = false;

    cout<<"Enter number of elements"<<"\n";
    cin>>iValue;

    Array aobj(iValue);
    aobj.Accept();

    bRet = aobj.SearchNum();
    if(bRet == true)
    {
        cout<<"Number is present"<<"\n";
    }
    else
    {
        cout<<"Number is not present"<<"\n";
    }

    return 0;
}